### PR TITLE
Mccalluc/cpu quota

### PR DESCRIPTION
**Pull request by mccalluc**
I don't know whether 1/2 a CPU is a reasonable default. I also should look a little bit harder at whether overall resource usage by docker engine can be controlled.

### CODE REVIEW

**Comment by mccalluc** [comment]
Failing on the new test:

`2.7201831340789795 not less than 0.24436116218566895` and
`2.628983974456787 not less than 0.8854141235351562`

Consistent enough to make me wonder if I reversed the order by mistake.

**Comment by mccalluc** [comment]
Earlier failing tests caused uniformly by a slow initial load of image. Thought about adding a cache warmup at class level, but I think it's more clear to put it in the test itself. This test is still fragile: CPU throttling, even at a 100:1 ratio, doesn't make a huge difference, particularly on travis?.... I think I'll try adding a lot of requests, or maybe also add an IO throttle?

@scottx611x : If you think I'm going in the wrong direction here, let me know?

File: tests/test_docker_utils.py [hunk 1]
```diff
@@ -136,46 +140,56 @@ def tearDown(self):
...
+
+        with Timer() as slow:
+            url = self.get_docker_url({'cpus': 0.01})
+            self.assert_loads_eventually(url, 'Welcome to nginx!')
+
+        with Timer() as fast_2:
+            url = self.get_docker_url({'cpus': 1})
+            self.assert_loads_eventually(url, 'Welcome to nginx!')
+
+        message = 'Less CPU should be slower: ' \
```
Comment (scottx611x) [inline tests/test_docker_utils.py]: Code level changes overall look good.

Maybe if these  tests are too inconsistent or if its too much work to diagnose discrepancies between local and Travis envs we could just test that the code assigning the different amounts of `nano_cpus` works and rely on docker to work as it states it should?

**Comment by mccalluc** [comment]
(run is now mocked, but real tmp dirs are still being made... but not cleaned up, because that's based on a list of running containers. --> Probably mock the directory creation step?)

**Comment by mccalluc** [comment]
@scottx611x : You might look at the latest changes, but I'll go ahead and merge.

**Pull request merged**